src/components/sets/SetsPage.jsx

```diff
@@ -1,17 +1,173 @@
 import { MainNavigation } from "../utils/MainNavigation.jsx";
-import { Col, Container, Row } from "react-bootstrap";
+import { Col, Container, Row, Spinner, Form } from "react-bootstrap";
+import { useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { useQuery } from "@apollo/client";
+import { GET_SETS_BY_CODE } from "../../assets/queries.jsx";
+import { ResponsiveLine } from "@nivo/line";
+import { ResponsivePie } from "@nivo/pie";
+import { ResponsiveBar } from "@nivo/bar";
+import {
+  getColorData,
+  getCardPriceData,
+  getRarityData,
+  getCardTypesData,
+} from "./SetsFunctions.jsx";
 
-export const SetsPage = () => {
+export const SetsPage = ({ setInfo, loadingSets: isLoading, errorSets: loadError }) => {
+  const [selectedSet, setSelectedSet] = useState("");
+  const [iconUri, setIconUri] = useState("");
+  const navigate = useNavigate();
+  if (loadError) {
+    navigate("/error", { state: { errorMessages: loadError.message } });
+  }
+
+  useEffect(() => {
+    if (setInfo.length > 0) {
+      const firstSet = setInfo[0];
+      setSelectedSet(firstSet.code); // Set the default selected set code
+      setIconUri(firstSet.icon_svg_uri); // Set the default icon URI
+    }
+  }, [setInfo]);
+
+  const handleSelectChange = (e) => {
+    const selectedCode = e.target.value;
+    setSelectedSet(selectedCode); // Update the selected set state
+
+    // Find the corresponding set from setInfo and update the iconUri
+    const selectedSetInfo = setInfo.find((set) => set.code === selectedCode);
+    if (selectedSetInfo) {
+      setIconUri(selectedSetInfo.icon_svg_uri);
+    }
+  };
+
+  //Load data from the API
+  const { loading, data } = useQuery(GET_SETS_BY_CODE(selectedSet));
+  if (loading) {
+    return "Loading...";
+  }
+
+  //Fetch data from functions
```

Comment:
I messed around this because if I remove your if (loading) code this will cause an error. The state management of the useQuery is confusing to me, but you can add a line like this: 
```const { loading, data } = useQuery(GET_SETS_BY_CODE(selectedSet), {
    onCompleted: (data) => {
      setColorData(getColorData(data));
      setPriceData(getCardPriceData(data));
      setRarityData(getRarityData(data));
      setTypesData(getCardTypesData(data));
    },
  });

There are probably other ways to handling it, but this was the only way I could get the spinner to load